msa/js-executor/queue/kafkaTemplate.js

```diff
@@ -33,21 +35,75 @@ let producer;
 
 const configEntries = [];
 
+let batchMessages = [];
+let batchResolvers = [];
+let sendLoopInstance;
+
 function KafkaProducer() {
-    this.send = async (responseTopic, scriptId, rawResponse, headers) => {
-        return producer.send(
-            {
-                topic: responseTopic,
-                acks: acks,
-                compression: compressionType,
-                messages: [
-                    {
-                        key: scriptId,
-                        value: rawResponse,
-                        headers: headers.data
-                    }
-                ]
-            });
+    this.send = (responseTopic, scriptId, rawResponse, headers) => {
+        logger.debug('Pending queue response, scriptId: [%s]', scriptId);
+        const message = {
+            topic: responseTopic,
+            messages: [{
+                key: scriptId,
+                value: rawResponse,
+                headers: headers.data
+            }]
+        };
+
+        return pushMessageToSendLater(message);
+    }
+}
+
+function pushMessageToSendLater(message) {
+    let resolver;
+    const promise = new Promise((resolve, reject) => {
+        resolver = resolve;
+    });
+    batchMessages.push(message);
```

Comment:
using both batchMessages and batchResolvers is not clear. Are you trying to save the amount of objects you create? Why not adding one object (wrapper) that contain both message and promise(resolver) and use an array of wrappers? This will make code cleaner.

Reply:
yes, it will be more clear, but this requires unwrapping for each message to produce the final array to send. so this approach is much lighter for the CPU.

Reply:
Ok